rskj-core/src/main/java/co/rsk/peg/BridgeStorageProvider.java

```diff
@@ -592,6 +602,89 @@ private void saveCoinbaseInformations() {
             safeSaveToRepository(getStorageKeyForCoinbaseInformation(blockHash), data, BridgeSerializationUtils::serializeCoinbaseInformation));
     }
 
+    public Long getActiveFederationCreationBlockHeight() {
+        if (!activations.isActive(RSKIP186)) {
+            return null;
+        }
+
+        if (activeFederationCreationBlockHeight != null) {
+            return activeFederationCreationBlockHeight;
+        }
+
+        activeFederationCreationBlockHeight = safeGetFromRepository(ACTIVE_FEDERATION_CREATION_BLOCK_HEIGHT_KEY, BridgeSerializationUtils::deserializeOptionalLong).orElse(null);
+        return activeFederationCreationBlockHeight;
+    }
+
+    public void setActiveFederationCreationBlockHeight(long activeFederationCreationBlockHeight) {
+        this.activeFederationCreationBlockHeight = activeFederationCreationBlockHeight;
+    }
+
+    protected void saveActiveFederationCreationBlockHeight() {
+        if (activeFederationCreationBlockHeight == null || !activations.isActive(RSKIP186)) {
+            return;
+        }
+
+        safeSaveToRepository(ACTIVE_FEDERATION_CREATION_BLOCK_HEIGHT_KEY, activeFederationCreationBlockHeight, BridgeSerializationUtils::serializeLong);
+    }
+
+    public Long getNextFederationCreationBlockHeight() {
+        if (!activations.isActive(RSKIP186)) {
+            return null;
+        }
+
+        if (nextFederationCreationBlockHeight != null) {
+            return nextFederationCreationBlockHeight;
+        }
+
+        nextFederationCreationBlockHeight = safeGetFromRepository(NEXT_FEDERATION_CREATION_BLOCK_HEIGHT_KEY, BridgeSerializationUtils::deserializeOptionalLong).orElse(null);
+        return nextFederationCreationBlockHeight;
+    }
+
+    public void setNextFederationCreationBlockHeight(long nextFederationCreationBlockHeight) {
+        this.nextFederationCreationBlockHeight = nextFederationCreationBlockHeight;
+    }
+
+    public void clearNextFederationCreationBlockHeight() {
+        this.nextFederationCreationBlockHeight = -1L;
```

Comment:
Maybe we should check if the activation is on

Reply:
not sure this is the responsibility of this class. It's data level entity - it simply stores and retrieves data.

Reply:
But, why to have a -1L. Which is the use case? A normal contract would retrieve 0 and nothing more (and manage the zero at code level). All the logic BEFORE the activation should live with a 0 in that data slot

Reply:
as discussed zero value doesn't clear it